github/data_source_github_repository.go

```diff
@@ -0,0 +1,163 @@
+			"homepage_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"private": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"has_issues": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"has_projects": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"has_downloads": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"has_wiki": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"allow_merge_commit": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"allow_squash_merge": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"allow_rebase_merge": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"default_branch": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"archived": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"topics": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"html_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ssh_clone_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"svn_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"git_clone_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"http_clone_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceGithubRepositoryRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+
+	orgName := meta.(*Organization).name
+	var repoName string
+
+	if fullName, ok := d.GetOk("full_name"); ok {
+		var err error
+		orgName, repoName, err = splitRepoFullName(fullName.(string))
+		if err != nil {
+			return err
+		}
+	}
+	if name, ok := d.GetOk("name"); ok {
+		repoName = name.(string)
+	}
+
+	if repoName == "" {
+		return fmt.Errorf("One of %q or %q has to be provided", "full_name", "name")
+	}
+
+	log.Printf("[DEBUG] Reading GitHub repository %s/%s", orgName, repoName)
+	repo, _, err := client.Repositories.Get(context.TODO(), orgName, repoName)
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", repoName)
+	d.Set("description", repo.Description)
+	d.Set("homepage_url", repo.Homepage)
+	d.Set("private", repo.Private)
+	d.Set("has_issues", repo.HasIssues)
+	d.Set("has_wiki", repo.HasWiki)
+	d.Set("allow_merge_commit", repo.AllowMergeCommit)
+	d.Set("allow_squash_merge", repo.AllowSquashMerge)
+	d.Set("allow_rebase_merge", repo.AllowRebaseMerge)
+	d.Set("has_downloads", repo.HasDownloads)
+	d.Set("full_name", repo.FullName)
+	d.Set("default_branch", repo.DefaultBranch)
+	d.Set("html_url", repo.HTMLURL)
+	d.Set("ssh_clone_url", repo.SSHURL)
+	d.Set("svn_url", repo.SVNURL)
+	d.Set("git_clone_url", repo.GitURL)
+	d.Set("http_clone_url", repo.CloneURL)
+	d.Set("archived", repo.Archived)
+	d.Set("topics", flattenStringList(repo.Topics))
+
+	d.SetId(repoName)
```

Comment:
_minor_ shouldn't this be above the d.Set()'s incase there's a crash?

Reply:
Good call, I'll move it.